public/js/utils/source-map.js

```diff
@@ -7,11 +7,16 @@ const { getSource } = require("../selectors");
 const { isEnabled } = require("../feature");
 const { isJavaScript } = require("./source");
 const assert = require("./assert");
+const { prefs } = require("./prefs");
 
 import type { Location } from "./actions/types";
 
 let sourceMapRequests = new Map();
 
+function shouldSourceMap() {
+  return isEnabled("sourceMaps") && prefs.sourceMapsEnabled;
```

Comment:
I don't really like having two pref systems. Eventually we'll have multiple duplicate prefs and need to do double-checks in lots of places.

I'd much rather go ahead and re-haul our config system to be based on prefs. `isEnabled` is supposed to be the thing that we can call directly without creating these functions.

I don't know what it looks like. Maybe our config system reads from prefs on startup, and also attaches pref observers so it can update the config when it's changed. What do you think?